pkg/skaffold/build/kaniko/sources/sources.go

```diff
@@ -61,35 +61,56 @@ func podTemplate(cfg *latest.KanikoBuild, args []string) *v1.Pod {
 					Image:           cfg.Image,
 					Args:            args,
 					ImagePullPolicy: v1.PullIfNotPresent,
-					Env: []v1.EnvVar{{
-						Name:  "GOOGLE_APPLICATION_CREDENTIALS",
-						Value: "/secret/kaniko-secret",
-					}},
-					VolumeMounts: []v1.VolumeMount{
-						{
-							Name:      constants.DefaultKanikoSecretName,
-							MountPath: "/secret",
-						},
-					},
+					Env:             []v1.EnvVar{},
+					VolumeMounts:    []v1.VolumeMount{},
 				},
 			},
 			RestartPolicy: v1.RestartPolicyNever,
-			Volumes: []v1.Volume{{
-				Name: constants.DefaultKanikoSecretName,
-				VolumeSource: v1.VolumeSource{
-					Secret: &v1.SecretVolumeSource{
-						SecretName: cfg.PullSecretName,
-					},
-				},
-			},
-			},
+			Volumes:       []v1.Volume{},
 		},
 	}
 
+	if cfg.GoogleCloudConfig == nil {
+		return addGoogleCloudConfig(pod, cfg.GoogleCloudConfig)
+	}
+
 	if cfg.DockerConfig == nil {
```

Comment:
```suggestion
	if cfg.DockerConfig != nil {
```
and we should cover this in a unit test too if possible